### PR TITLE
Bugfix/header values composing

### PR DESCRIPTION
**Pull request by coutoPL**
Ref #366

Soap envelope should be:

```
<soap12:Envelope>
    <soap12:Header>
      <sessionId>session</sessionId>
      <correlationId>10001</correlationId>
    </soap12:Header>
    <soap12:Body>
      <username>User</username>
    </soap12:Body>
</soap12:Envelope>
```

not:

```
<soap12:Envelope>
    <soap12:Header>
      session10001
    </soap12:Header>
    <soap12:Body>
      User
    </soap12:Body>
</soap12:Envelope>
```

for tested WSDLs (`implicit_header_example.wsdl`, `explicit_header_example.wsdl` and `implicit_header_multiple_part_header.wsdl`)

### CODE REVIEW

File: integration/src/test/scala/Wsdl11Soap12Test.scala [hunk 1]
```diff
@@ -38,14 +38,18 @@ object Wsdl11Soap12Test extends TestBase {
         |val service = new implicitheader.UserBindings with scalaxb.SoapClients with scalaxb.HttpClients {
         |      override def httpClient = new HttpClient {
         |        override def request(in: String, address: java.net.URI, headers: Map[String, String]): String = {
+        |        println("!!!" + in)
```
Comment (eed3si9n) [inline integration/src/test/scala/Wsdl11Soap12Test.scala]: !!! :)

File: cli/src/main/scala/scalaxb/compiler/wsdl11/GenSource.scala [hunk 2]
```diff
@@ -300,9 +300,9 @@ trait {interfaceTypeName} {{
 
   private def implicitHeaderParts(headers: Seq[HeaderBinding]): Seq[XPartType] =
     headers.flatMap { header =>
-      context.messages(splitTypeName(header.message))
-      .part.filter(_.name == Some(header.part))
-      .map(x => x.copy(name = x.name.map(camelCase)))
+      context.messages(splitTypeName(header.message)).part
+        .filter(_.name.contains(header.part))
```
Comment (eed3si9n) [inline cli/src/main/scala/scalaxb/compiler/wsdl11/GenSource.scala]: Does this change the behavior?

Comment (coutoPL) [inline cli/src/main/scala/scalaxb/compiler/wsdl11/GenSource.scala]: no ... IMO it's just cleaner

**Comment by eed3si9n** [comment]
Thanks for following up with a quick fix!

File: cli/src/main/scala/scalaxb/compiler/wsdl11/GenSource.scala (hunk 2 above)
Comment (coutoPL) [inline cli/src/main/scala/scalaxb/compiler/wsdl11/GenSource.scala]: ok, but there is no contain in Option class of scala 2.10. Reverted.

**Comment by eed3si9n** [comment]
I don't know how Apache CXF encodes implicit headers, but if it's supported, it might be worth running the server and hitting it with scalaxb like here:
- https://github.com/eed3si9n/scalaxb/blob/master/integration/src/test/scala/DocumentLiteralBareService.scala
- https://github.com/eed3si9n/scalaxb/blob/master/integration/src/test/scala/Wsdl11DocumentBareTest.scala

**Comment by coutoPL** [comment]
I've tried to force apache CXF to generate wsdl with implicit headers and unfortunately I couldn't. Google says nothing about using implicit headers by CXF, so I guess it's impossible to achieve.

**Comment by eed3si9n** [comment]
I'm going to restart Travis to see what happens. I think I've seen it fail like that before.

**Pull request merged**